File: 7 kyu/You're a square!/solution.ex
defmodule Math do
  def square?(n) when n < 0, do: false
   
   def square?(n), do: :math.sqrt(n) == round(:math.sqrt(n))
end
__________________________________
defmodule Math do
  def square?(n) when n < 0, do: false
  def square?(n), do: trunc(:math.sqrt(n)) == :math.sqrt(n)
end
__________________________________
defmodule Math do
    def square?(n) when is_number(n) and n>=0 do
        Float.floor(:math.sqrt(n)) == :math.sqrt(n)
    end
    def square?(n), do: false
end
__________________________________
defmodule Math do
  def square?(n) do
      square?(n,0)
    end

    defp square?(n,i) when (i * i) == n, do: true

    defp square?(n,i) when (i * i) < n, do: square?(n,i+1)

    defp square?(n,i) when (i * i) > n, do: false
end
__________________________________
defmodule Math do
  def square?(n) when n < 0, do: false  
  def square?(n) do
    Enum.any?(1..n, fn x -> x * x == n end)
  end
end

File: 7 kyu/You're a square!/solution.cpp
#include<cmath>

bool is_square(int n) {
  if (n < 0) return false;
  int square = sqrt(n);
  return square * square == n;
}
__________________________________
#include <cmath>

bool is_square(int n)
{
  return fmod(sqrt(n), 1) == 0;
}
__________________________________
bool is_square(int n)
{
  for (int i = 0; i <= n/2 + 1; i++){
  if ( i * i == n)
  return true;
  }
  return false;
}
__________________________________
#include <math.h>
bool is_square(int n)
{
  double x;
  if(n<0) return false;
  x=sqrt(n);
  if(int(x)!=x) return false;
  return true;
}
